x-pack/test/observability_ai_assistant_api_integration/tests/conversations/conversations.spec.ts

```diff
@@ -250,5 +253,85 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         });
       });
     });
+
+    describe('security roles and access privileges', () => {
+      let createResponse: Awaited<
+        SupertestReturnType<'POST /internal/observability_ai_assistant/conversation'>
+      >;
+
+      before(async () => {
+        createResponse = await observabilityAIAssistantAPIClient
+          .editor({
+            endpoint: 'POST /internal/observability_ai_assistant/conversation',
+            params: {
+              body: {
+                conversation: conversationCreate,
+              },
+            },
+          })
+          .expect(200);
+      });
+
+      const routes: Array<{
+        endpoint: string;
+        method: 'get' | 'post' | 'put' | 'delete';
+        payload?: Record<string, any>;
+        requiredPrivileges: string[];
+      }> = [
+        {
+          endpoint: '/internal/observability_ai_assistant/conversations',
+          method: 'post',
+          requiredPrivileges: ['ai_assistant'],
+        },
+        {
+          endpoint: '/internal/observability_ai_assistant/conversation/{conversationId}',
+          method: 'put',
+          payload: {
+            path: { conversationId: 'test-id' },
+            body: { conversation: conversationUpdate },
+          },
+          requiredPrivileges: ['ai_assistant'],
+        },
+        {
+          endpoint: '/internal/observability_ai_assistant/conversation/{conversationId}',
+          method: 'get',
+          payload: {
+            path: { conversationId: 'test-id' },
+          },
+          requiredPrivileges: ['ai_assistant'],
+        },
+        {
+          endpoint: '/internal/observability_ai_assistant/conversation/{conversationId}',
+          method: 'delete',
+          payload: {
+            path: { conversationId: 'test-id' },
+          },
+          requiredPrivileges: ['ai_assistant'],
+        },
+      ];
+
+      routes.forEach((route) => {
+        const { endpoint, method, payload } = route;
+
+        describe(`${method.toUpperCase()} ${endpoint}`, () => {
+          it('should deny access for unauthorized users', async () => {
+            const request = supertestWithoutAuth[method](
```

Comment:
As mentioned on @arturoliduena's [PR](https://github.com/elastic/kibana/pull/200137/files#r1868492703) I suggest we use `observabilityAIAssistantAPIClient` for calling our own endpoints.

Reply:
Sounds good, Thanks @sorenlouv 
I've updated the code